src/input/SimModSuite.h

```diff
@@ -309,6 +329,314 @@ class SimModSuite : public MeshInput
+       {
+          mycube = *it;
+          logInfo(PMU_rank()) << "Cube mesh refinement: " <<mycube.CubeMSize;
+          logInfo(PMU_rank()) << "Center"<< mycube.CubeCenter[0]<<" "<<mycube.CubeCenter[1]<<" "<<mycube.CubeCenter[2];
+          logInfo(PMU_rank()) << "Width"<< mycube.CubeWidth[0]<<" "<<mycube.CubeWidth[1]<<" "<<mycube.CubeWidth[2];
+          logInfo(PMU_rank()) << "Height"<< mycube.CubeHeight[0]<<" "<<mycube.CubeHeight[1]<<" "<<mycube.CubeHeight[2];
+          logInfo(PMU_rank()) << "Depth"<< mycube.CubeDepth[0]<<" "<<mycube.CubeDepth[1]<<" "<<mycube.CubeDepth[2];
+          MS_addCubeRefinement (meshCase, mycube.CubeMSize, &mycube.CubeCenter[0], &mycube.CubeWidth[0], &mycube.CubeHeight[0], &mycube.CubeDepth[0]);
+       }
+    }
+
+    if (MeshAtt.MeshSizePropagationDistance>0.0) {
+       std::list<int>::iterator it;
+       for (it=MeshAtt.lFaceIdMeshSizePropagation.begin(); it != MeshAtt.lFaceIdMeshSizePropagation.end(); it++)
+       {
+         pGEntity face = GM_entityByTag(model, 2, *it);
+         if (face == NULL) {
+           logError() << "MeshSizeProp faceid:"<<*it <<"not found in model.";
+         } else {
+           logInfo(PMU_rank()) << "MeshSizeProp faceid:"<<*it <<", distance =" << MeshAtt.MeshSizePropagationDistance << ", scaling factor =" << MeshAtt.MeshSizePropagationScalingFactor;
+#ifdef BEFORE_SIM_15
+           MS_setMeshSizePropagation(meshCase,face,1,MeshAtt.MeshSizePropagationDistance,MeshAtt.MeshSizePropagationScalingFactor);
+#else
+           MS_setMeshSizePropagation(meshCase,face,2,1,MeshAtt.MeshSizePropagationDistance,MeshAtt.MeshSizePropagationScalingFactor);
+#endif
+           }
+       }
+    }
+     if (MeshAtt.lFaceIdUseDiscreteMesh.size()>0) {
+       std::list<int>::iterator it;
+       for (it=MeshAtt.lFaceIdUseDiscreteMesh.begin(); it != MeshAtt.lFaceIdUseDiscreteMesh.end(); it++)
+       {
+           pGEntity face = GM_entityByTag(model, 2, *it);
+           if (face == NULL) {
+            logError() << "UseDiscreteMesh; faceid:"<<*it <<"not found in model.";
+           } else {
+             logInfo(PMU_rank()) << "UseDiscreteMesh; faceid, noModification:"<<*it, MeshAtt.UseDiscreteMesh_noModification;
+             MS_useDiscreteGeometryMesh(meshCase,face,MeshAtt.UseDiscreteMesh_noModification);
+           //MS_limitSurfaceMeshModification(meshCase,face,UseDiscreteMesh_noModification);
+           }
+       }
+    } 
+}
+
+
+private:
+void loadCAD(const char* modFile, const char* cadFile){
+    // Load CAD
+    std::string sCadFile;
+    if (cadFile)
+        sCadFile = cadFile;
+    else {
+        sCadFile = modFile;
+        utils::StringUtils::replaceLast(sCadFile, ".smd", "_nat.x_t");
+    }
+    pNativeModel nativeModel = 0L;
+    if (utils::Path(sCadFile).exists())
+        nativeModel = ParasolidNM_createFromFile(sCadFile.c_str(), 0);
+
+    m_model = GM_load(modFile, nativeModel, 0L);
+
+    if (nativeModel)
```

Comment:
I thought native Models were removed in #16 ?

Reply:
see above comment. No it should not be removed but we have no file to test